Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
+					m_GoldSlider->SetValue(selectedGA->GetDefaultGoldCake());
+				} else if (m_DifficultySlider->GetValue() < Activity::EasyDifficulty && selectedGA->GetDefaultGoldEasy() > -1) {
+					m_GoldSlider->SetValue(selectedGA->GetDefaultGoldEasy());
+				} else if (m_DifficultySlider->GetValue() < Activity::MediumDifficulty && selectedGA->GetDefaultGoldMedium() > -1) {
+					m_GoldSlider->SetValue(selectedGA->GetDefaultGoldMedium());
+				} else if (m_DifficultySlider->GetValue() < Activity::HardDifficulty && selectedGA->GetDefaultGoldHard() > -1) {
+					m_GoldSlider->SetValue(selectedGA->GetDefaultGoldHard());
+				} else if (m_DifficultySlider->GetValue() < Activity::NutsDifficulty && selectedGA->GetDefaultGoldNuts() > -1) {
+					m_GoldSlider->SetValue(selectedGA->GetDefaultGoldNuts());
+				} else if (selectedGA->GetDefaultGoldNuts() > -1) {
+					m_GoldSlider->SetValue(selectedGA->GetDefaultGoldNuts());
+				}
+				m_GoldSlider->SetEnabled(selectedGA->GetGoldSwitchEnabled());
+
+				// Set default checkbox states and enable or disable them.
+				int defaultFogOfWar = selectedGA->GetDefaultFogOfWar();
+				if (defaultFogOfWar > -1) {
+					m_FogOfWarCheckbox->SetCheck(defaultFogOfWar != 0);
+				}
+				m_FogOfWarCheckbox->SetEnabled(selectedGA->GetFogOfWarSwitchEnabled());
+
+				int defaultReqClearPath = selectedGA->GetDefaultRequireClearPathToOrbit();
+				if (defaultReqClearPath > -1) {
+					m_RequireClearPathToOrbitCheckbox->SetCheck(defaultReqClearPath != 0);
+				}
+				m_RequireClearPathToOrbitCheckbox->SetEnabled(selectedGA->GetRequireClearPathToOrbitSwitchEnabled());
+
+				int defaultDeployUnits = selectedGA->GetDefaultDeployUnits();
+				if (defaultDeployUnits > -1) {
+					m_DeployUnitsCheckbox->SetCheck(defaultDeployUnits != 0);
+				}
+				m_DeployUnitsCheckbox->SetEnabled(selectedGA->GetDeployUnitsSwitchEnabled());
 			}
-			m_pGoldSlider->SetEnabled(pSelectedGA->GetGoldSwitchEnabled());
-
-			//Set default fog of war flag and enable or disable it if necessary
-			if (pSelectedGA->GetDefaultFogOfWar() > -1)
-			{
-				if (pSelectedGA->GetDefaultFogOfWar() == 0)
-					m_pFogOfWarCheckbox->SetCheck(0);
-				else
-					m_pFogOfWarCheckbox->SetCheck(1);
+		} else {
+			m_ScenarioScenes = nullptr;
+			m_ActivityLabel->SetText("No Activity selected.");
+			m_DifficultyLabel->SetVisible(false);
+			if (m_DifficultySlider) {
+				m_DifficultySlider->SetVisible(false);
 			}
-			m_pFogOfWarCheckbox->SetEnabled(pSelectedGA->GetFogOfWarSwitchEnabled());
-
-			//Set default clear path to orbit flag and enable or disable it if necessary
-			if (pSelectedGA->GetDefaultRequireClearPathToOrbit() > -1)
-			{
-				if (pSelectedGA->GetDefaultRequireClearPathToOrbit() == 0)
-					m_pRequireClearPathToOrbitCheckbox->SetCheck(0);
-				else
-					m_pRequireClearPathToOrbitCheckbox->SetCheck(1);
+
+			int textHeight = m_ActivityLabel->ResizeHeightToFit();
```

Comment:
These variables can really just be inlined into the function call, or at least merged into one variable